### PR TITLE
git-workflow: Add info on good commits

### PR DESCRIPTION
**Pull request by zeenix**

### CODE REVIEW

**Review by GordanM — APPROVED** [review]

**Review by rpannek — APPROVED** [review]

File: docs/chapters/workflow/git-workflow.rst [hunk 1]
```diff
@@ -65,3 +65,26 @@ force-pushed to the forked repo, or they can be added on top of the existing com
...
+
+Good commits
+============
+
+A good commit (or patch):
+
+#. is atomic: Each commit should contain only one logical change. Use of vague verbs (e.g "update")
+   and phrases like "this patch also" are usually a good indicator that commit is not atomic.
+
+#. has a good commit message associated with it.
```
Comment (DunderRoffe) [inline docs/chapters/workflow/git-workflow.rst:77]: There needs to be an explanation or at least a link to an explanation of what we believe a good commit message is. Furthermore, a good commit message is a bit to vague. We have rules of which some are not trivial which contributors never will guess on their own.

**Review by DunderRoffe — CHANGES_REQUESTED** [review]

File: docs/chapters/workflow/git-workflow.rst (hunk 1 above)
Comment (DunderRoffe) [reply docs/chapters/workflow/git-workflow.rst:77]: OK saw that you have a link below. However, that is not clear and I stand by my point that the above line is confusing without any description of what a good commit message is. Good commit implies good commit message sounds awfully trivial

Comment (zeenix) [reply docs/chapters/workflow/git-workflow.rst:77]: The rationale below gives a good (even though vague) guideline on what a good commit message would be. For more specific example, readers can follow the link provided.

**Review by DunderRoffe — APPROVED** [review]
Well I guess it is good enough...

**Pull request merged**